src/App.vue

```diff
@@ -0,0 +1,343 @@
+												@click="saveSettings(mountidx, $event)" />&nbsp;&nbsp;&nbsp;{{
+														saveSettingsLabel[mountidx]
+												}}
+										</td>
+									</tr>
+								</tbody>
+							</table>
+						</div>
+
+						<div style="border-bottom: 1px solid #ccc !important; padding: 20px 20px 20px 20px;"></div>
+
+						<div><u>Purchase new Stamp:</u></div>
+
+						<div>
+							<form @submit.prevent>
+								<table id="" style="border-style: solid;">
+									<thead>
+										<tr>
+											<th>Amount:</th>
+											<th>Depth</th>
+											<th></th>
+										</tr>
+									</thead>
+									<tbody>
+										<tr>
+											<td><input v-model="newBatchAmounts[mountidx]" type="number" value=""
+													maxlength="10" /></td>
+											<td><input v-model="newBatchDepths[mountidx]" type="number" value=""
+													maxlength="17" />
+
+											</td>
+											<td>
+												<input type="submit" :disabled="newBatchBtnDisabled[mountidx]"
+													value="Buy"
+													@click="buyPostage(mountidx, $event)" />&nbsp;&nbsp;&nbsp;{{
+															newBatchLabel[mountidx]
+													}}
+												<br />
+												<br />
+											</td>
+										</tr>
+									</tbody>
+								</table>
+							</form>
+						</div>
+						<div style="border-bottom: 2px solid #ccc !important; padding: 20px 20px 20px 20px;"></div>
+					</div>
+				</div>
+			</div>
+		</div>
+	</AppContent>
+</template>
+
+<script>
+/* eslint-disable no-console */
+import AppContent from "@nextcloud/vue/dist/Components/AppContent";
+import Actions from '@nextcloud/vue/dist/Components/Actions';
+import ActionButton from '@nextcloud/vue/dist/Components/ActionButton';
+import CheckboxRadioSwitch from '@nextcloud/vue/dist/Components/CheckboxRadioSwitch';
+import ActionInput from '@nextcloud/vue/dist/Components/ActionInput';
+import ActionSeparator from '@nextcloud/vue/dist/Components/ActionSeparator';
+import { BeeDebug } from "@ethersphere/bee-js";
+import axios from "axios";
+import { generateUrl } from '@nextcloud/router';
+
+export default {
+	name: "App",
+	components: {
+		AppContent, Actions, ActionButton, CheckboxRadioSwitch, ActionInput, ActionSeparator
+	},
+	props: {
+		settings: {
+			type: Object,
+			default() {
+				return {};
+			},
+		},
+	},
+	data() {
+		// Here the component data is stored with inital values
+		return {
+			beeClient: null,
+			show: true,
+			parsedMounts: null,
+			newBatchAmounts: [],
+			newBatchDepths: [],
+			newBatchLabel: [],
+			newBatchBtnDisabled: [],
+			toggleNode: [],
+			saveSettingsValue: [],
+			saveSettingsBtn: [],
+			saveSettingsLabel: [],
+		};
+	},
+	computed: {
+	},
+	async mounted() {
+		// Code in here runs before component is mounted to DOM
+		this.parsedMounts = JSON.parse(this.settings.mounts);
+		this.parsedMounts = this.parsedMounts.map(mount => ({ ...mount, isEncrypted: mount.encrypt === 1, batches: mount.batches.map(b => ({ ...b, topUpValue: 0 })) }));
```

Comment:
This structure is quite hard to work with. I cannot see any major reason to restructure it but if the parseMounts structure changes later this will be a headache. No suggestion from me yet. Keep it as it.